docs/reference/modules/widget-type.md

```diff
@@ -0,0 +1,179 @@
+
+### `contextual`
+
+When the `contextual` option is set to `true`, an edit button is not displayed for the widget, and the widget is not edited in a dialog box according to a schema of fields. Instead, the widget is edited directly on the page.
+
+To support this, the widget must implement the [contextual editing pattern](../../guide/editing-custom-widgets-in-context.md).
+
+Defaults to `false`.
+
+### `defaultData`
+
+The `defaultData` option may contain an object providing defaults for contextually edited widgets. This is required for contextually edited widgets because they do not always use the `fields` section. Normal widgets that use the usual dialog box to edit schema fields may rely on the `def` property of each field instead. There is no default value.
+
+### `deferred`
+
+Widget types may contain `relationship` and `relationshipReverse` fields that load other documents. By default this is done in the order the widgets are encountered when loading the page, with requests relating to the same type of widget grouped together for performance. This is a recursive process. Widget types that specify `deferred: true` will resolve their "loaders" at the very end of the process, after all other widget types and their relationships have been recursively loaded. This is beneficial for widget type modules like `@apostrophecms/image-widget` because they can be efficiently grouped together in a single database call. Defaults to `false`.
+
+### `icon`
+
+The name of the icon to be displayed for this widget type in a menu of widget types. This icon name must correspond to an icon loaded via [the `icons` module section](/reference/module-api/module-overview.md#icons). If not configured the widget type is listed without an icon.
+
+### `label`
+
+The label to be displayed for the icon in a menu of widget types and in certain other contexts in the UI. This should be brief but informative and should be capitalized, like `Slideshow`. By default it is derived from the module name, but we recommend configuring an explicit label.
+
+### `neverLoad`
+
+This option specifies an array of widget types that should never be loaded recursively by this widget type. While documents that contain those widget types might be loaded by relationships, additional relationships within those widget types will not be loaded. This can be a helpful guard against runaway recursion and the associated performance hit. There is no default, however see [`neverLoadSelf`](#never-load-self).
+
```

Comment:
I understand what this does and why it may be helpful, but from this definition I don't have much of a hint about _when_ I'd want to use it. Is there a succinct example to include?